files/zh-cn/web/http/basics_of_http/choosing_between_www_and_non-www_urls/index.md

```diff
@@ -9,51 +9,53 @@ slug: Web/HTTP/Basics_of_HTTP/Choosing_between_www_and_non-www_URLs
-在这种情况下，你需要配置服务器接收的 HTTP 请求（常见为 www 和非 www 网址相同）以及适当的 HTTP 响应 {{HTTPStatus(301)}} 去响应所有非规范的域名请求。这会将尝试使访问非规范网址的浏览器重定向到其规范的等效网址。举例来说，如果您选择使用非 www 网址为规范类型，你的所有 www 网址都应该被重定向到对应的非 www 网址上。
+在这种情况下，你需要配置接收 HTTP 请求的那台服务器（一般来说 www 和非 www 网址在同一台），让其用恰当的 HTTP {{HTTPStatus(301)}} 响应来应答所有发往非规范的域名的请求。这会将尝试使访问非规范网址的浏览器重定向到其规范的等效网址。例如，选择了非 www 网址作为规范网址后，所有向 www 网址的请求，都应该被重定向到不含 www 的等效网址。
 
 例如：
 
-1. 服务器收到 `http://www.example.org/whaddup` 请求（当规范域名是 example.org 时）
-2. 服务器则以代码 {{HTTPStatus(301)}} 与头 {{HTTPHeader("Location")}} ：`http://example.org/whaddup`
-3. 该客户端发出的规范的域名请求：`http://example.org/whaddup`
+1. 服务器收到向 `http://www.example.org/whaddup` 的请求（当规范域名是 example.org 时）。
+2. 服务器回应代码 {{HTTPStatus(301)}}，以及标头 `{{HTTPHeader("Location")}}: http://example.org/whaddup`。
+3. 该客户端向规范域名下的地址——`http://example.org/whaddup`——发送请求。
 
-[HTML5 boilerplate project](https://github.com/h5bp/html5-boilerplate) 有一个示例 [how to configure an Apache server to redirect one domain to the other](https://github.com/h5bp/html5-boilerplate/blob/7a22a33d4041c479d0962499e853501073811887/.htaccess#L219-L258)。
+[HTML5 boilerplate project](https://github.com/h5bp/html5-boilerplate) 中的一个例子演示了[如何配置一台 Apache 服务器将一个域名重定向到另一个](https://github.com/h5bp/html5-boilerplate/blob/7a22a33d4041c479d0962499e853501073811887/.htaccess#L219-L258)。
 
 ### 使用 \<link rel="canonical">
 
-它可以将一个特殊的 HTML {{HTMLElement("link")}} 元素添加到网页指示什么网页的标准地址，这对页面的访问者没有影响，但在搜索引擎检索时会告诉搜索引擎当页面实际的地址。通过这种方式，搜索引擎不需要索引同一页面多次，那样可能导致它被视为重复的内容或垃圾邮件，甚至从搜索引擎结果中删除或者降低你的页面显示排名。
+将一个特殊的 HTML {{HTMLElement("link")}} 元素添加到网页，就能指明网页的规范地址。它对页面的普通读者没有影响，但会告诉搜索引擎爬虫这个页面实际在哪。这样一来，搜索引擎不会对同一页面索引多次——这可能导致页面被视为重复内容或垃圾内容，甚至于搜索引擎结果中会删除你的页面或者降低你的排名。
 
-当加入这样一个标签，会告诉搜索引擎，你提供相同内容的两个域名那一个是规范的。以前面的例子为例，`http://www.example.org/whaddup` 将提供与 `http://example.org/whaddup` 相同的内容，但有一个附加的 {{HTMLElement("link")}} 头部元素：
+当加入这个标签，在两个域名中提供相同的内容，会告诉搜索引擎哪个域名是规范的。以前面的例子为例，`http://www.example.org/whaddup` 将提供与 `http://example.org/whaddup` 相同的内容，但有一个额外的 {{HTMLElement("link")}} 头部元素：
 
-`<link href="http://example.org/whaddup" rel="canonical">`
+```html
+<link href="http://example.org/whaddup" rel="canonical" />
+```
 
-不同于以往，浏览器历史记录将考虑非 www 和 www 的网址作为独立的条目。
+与前一种方法不同的是，在浏览器历史记录中非 www 和 www 的网址是独立的条目。
 
 ## 使你页面都有效
 
-有了这些技术，你可以配置服务器对 www 前缀和非 www 前缀的域名进行正确的响应，如果你无法预测用户会在浏览器地址栏使用哪个 URL，你需要做的就是选择一个作为你的规范地址，然后重定向另一个。
+有了这些技术，你可以配置服务器对 www 前缀和非 www 前缀的域名进行正确的响应。你无法预测用户会在浏览器地址栏输入哪个 URL，你需要做的就是选择一个作为你的规范地址，然后将另一个地址重定向到它。
 
 ## 根据情况决定
 
-可以认为这是一个非常主观 [bikeshedding](http://bikeshed.com/) 问题。如果你想更深入的阅读，请参阅 [WWW vs non-WWW for your Canonical Domain URL – Which is Best and Why?](http://www.hyperarts.com/blog/www-vs-non-www-for-your-canonical-domain-url-which-is-best-and-why/) ，它可能提出进一步的见解。
+可以认为这是一个非常主观的 [bikeshedding](http://bikeshed.com/) 问题。如果你想更深入地阅读，有 [很多](https://www.netlify.com/blog/2020/03/26/how-to-set-up-netlify-dns-custom-domains-cname-and-a-records/#options-for-bare-domains) [文章](https://www.wpbeginner.com/beginners-guide/www-vs-non-www-which-is-better-for-wordpress-seo/) 可供你参考。
```

Comment:
I've create a PR in mdn/content (mdn/content/pull/28543) to fix the description. So let's wait before I could correct this and go to merge it :)

Reply:
```suggestion
可以认为这是一个非常主观的[自行车棚效应](http://bikeshed.com/)。如果你想更深入地阅读，这里有一些资源：

- www.netlify.com 上的[使用裸域名的选项](https://www.netlify.com/blog/2020/03/26/how-to-set-up-netlify-dns-custom-domains-cname-and-a-records/#options-for-bare-domains)（2020）
- www.wpbeginner.com 上的 [WWW 和非 WWW——哪个更适合 WordPress 的 SEO？](https://www.wpbeginner.com/beginners-guide/www-vs-non-www-which-is-better-for-wordpress-seo/)（2023）

```